analyzer/analyzer.go

```diff
@@ -16,23 +16,41 @@ import (
 	"golang.org/x/tools/go/analysis"
 )
 
+type optionInt struct {
+	global  bool
+	intConv bool
+}
+
+type optionErr struct {
+	global   bool
+	errError bool
+	errorf   bool
+}
+
+type optionStr struct {
+	global    bool
+	sprintf1  bool
+	strconcat bool
+}
+
 type perfSprint struct {
-	intConv    bool
-	errError   bool
-	errorf     bool
-	sprintf1   bool
+	intFormat optionInt
+	errFormat optionErr
+	strFormat optionStr
+
+	boolFormat bool
+	hexFormat  bool
 	fiximports bool
-	strconcat  bool
 }
 
 func newPerfSprint() *perfSprint {
 	return &perfSprint{
-		intConv:    true,
-		errError:   false,
-		errorf:     true,
-		sprintf1:   true,
+		intFormat:  optionInt{global: true, intConv: true},
+		errFormat:  optionErr{global: true, errError: false, errorf: true},
```

Comment:
The errError default value was already false before this refactoring.

OK, so it requires to use --err-error=true to activate it.

It doesn't change with this refactoring, so it's good.

But then, there is something strange with inheritance, I'm not sure how to call it. At least, the fact one option depends on the others.

But, now let's assume someone use --err-format=true won't they expect the errError flag to be true by doing so

Reply:
I suggested a different approach that would avoid the confusion I hope

https://github.com/catenacyber/perfsprint/pull/41#discussion_r1935383659

Reply:
handled below